src/cwt/claim.rs

```diff
@@ -0,0 +1,352 @@
+use serde::{Deserialize, Serialize};
+use serde_cbor::Value;
+use time::OffsetDateTime;
+
+/// Representation of CWT claims, as defined in
+/// [RFC8392](https://datatracker.ietf.org/doc/html/rfc8392).
+pub trait Claim: Into<Value> + TryFrom<Value, Error = Error> {
+    fn key() -> Key;
+}
+
+/// Representation of the CBOR map key used to identify a claim
+/// within a CWT claims set, and restricted to text and integer values,
+/// per [RFC8392](https://datatracker.ietf.org/doc/html/rfc8392).
+pub enum Key {
+    Text(String),
+    Integer(i128),
+}
+impl From<Key> for Value {
+    fn from(key: Key) -> Value {
+        match key {
+            Key::Text(k) => Value::Text(k),
+            Key::Integer(k) => Value::Integer(k),
+        }
+    }
+}
+
+/// Errors that can occur when parsing values into claims.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
```

Comment:
is this overkill? can you see a better way of keeping things generic so others can implement traits that can be used with `get_claims`?